sympy/functions/elementary/hyperbolic.py

```diff
@@ -189,12 +189,24 @@ def _eval_as_leading_term(self, x):
             return self.func(arg)
 
     def _eval_is_real(self):
-        return self.args[0].is_real
-
-    def _eval_is_finite(self):
+        from sympy import pi
         arg = self.args[0]
+        if arg.is_real:
+            return True
         if arg.is_imaginary:
+            return False
```

Comment:
sinh(iy) = i sin(y) is zero when y is a multiple of pi. Hence this is not correct.